src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -0,0 +1,902 @@
+     * @property waypoints
+     * @type {array<WaypointModel>}
+     */
+    get waypoints() {
+        return _reduce(this._legCollection, (waypointList, legModel) => {
+            return waypointList.concat(legModel.waypoints);
+        }, []);
+    }
+
+    // ------------------------------ LIFECYCLE ------------------------------
+
+    /**
+     * Initialize class properties
+     *
+     * @for RouteModel
+     * @method init
+     * @param routeString {string}
+     * @chainable
+     */
+    init(routeString) {
+        this._legCollection = this._generateLegsFromRouteString(routeString);
+
+        return this;
+    }
+
+    /**
+     * Reset class properties
+     *
+     * @for RouteModel
+     * @method reset
+     * @chainable
+     */
+    reset() {
+        this._legCollection = [];
+
+        return this;
+    }
+
+    // ------------------------------ PUBLIC ------------------------------
+
+    // FIXME: COMPLETE THIS METHOD
+    /**
+     * Merge the provided route model into this route model, if possible
+     *
+     * @for RouteModel
+     * @method absorbRouteModel
+     * @param routeModel {RouteModel}
+     */
+    absorbRouteModel(/* routeModel */) {
+        //
+    }
+
+    /**
+     * Calculate the heading from the first waypoint to the second waypoint
+     *
+     * This is used to determine the heading of newly spawned aircraft
+     *
+     * @for RouteModel
+     * @method calculateSpawnHeading
+     * @return {number} heading, in radians
+     */
+    calculateSpawnHeading() {
+        const firstWaypointPositionModel = this.waypoints[0].positionModel;
+        const secondWaypointPositionModel = this.waypoints[1].positionModel;
+        const heading = firstWaypointPositionModel.bearingToPosition(secondWaypointPositionModel);
+
+        return heading;
+    }
+
+    /**
+    * Return an array of waypoints in the flight plan that have altitude restrictions
+    *
+    * @for RouteModel
+    * @method getAltitudeRestrictedWaypoints
+    * @return {array<WaypointModel>}
+    */
+    getAltitudeRestrictedWaypoints() {
+        return this.waypoints.filter((waypoint) => waypoint.hasAltitudeRestriction);
+    }
+
+    /**
+     * Return the ICAO identifier for the airport at whose runway this route will terminate
+     *
+     * @for LegModel
+     * @method getArrivalRunwayAirportIcao
+     * @return {string}
+     */
+    getArrivalRunwayAirportIcao() {
+        if (!this.hasStarLeg()) {
+            return null;
+        }
+
+        const starLegIndex = this._findStarLegIndex();
+
+        return this._legCollection[starLegIndex].getArrivalRunwayAirportIcao();
+    }
+
+    /**
+     * Return the `AirportModel` at whose runway this route will terminate
+     *
+     * @for RouteModel
+     * @method getArrivalRunwayAirportModel
+     * @return {AirportModel}
+     */
+    getArrivalRunwayAirportModel() {
+        const airportIcao = this.getArrivalRunwayAirportIcao();
+
+        if (airportIcao === null) {
```

Comment:
two things here:
1) is `airportIcao` a string?
    1a) if yes, then it should never be `null`. it should be an empty string
2) if `null` is in fact an appropriate data type, this should instead check for `!airportIcao` instead of `null`

doing this:
```javascript
airportIcao === null
```
is not the same as
```javascript
!airportIcao
```

yay javascript!

Reply:
1. It originally comes from `._getArrivalRunwayAirportModel()` called on line 256 in the preview thing above and when that finds nothing, it returns null. So this if block is checking to see if we failed to find an airport ICAO identifier in any leg's route strings. So perhaps an odd way to be doing it, but the logic aligns correctly. Now changed to at least use `!airportIcao` and will note that for later 😄